### PR TITLE
Ansible python bootstrap

### PR DESCRIPTION
**Pull request by gitschaub**
1. Refactor bootstrap- Bootstrap.sh is used to get a python interpreter on coreos. I would like to move this to ansible code. Since ansible is severely limited w/o python, you need to use the ansible raw module that essentially delivers shell commands over SSH to the host. Converting the script to a series of RAW tasks is still better than using a script. Just make sure you implement equivalent or better functionality.

Addressed by da393a7
1. Migrate [2] from the cloud-config yml to ansible bootstrap. Bin’s are stored in this directory for CoreOS b/c /usr/bin, /usr/local/bin, etc.. are read-only in CoreOS. I implemented the directory creation a while ago in the cloud-config yml as a quick fix. We want to move this to the pre-ansible role for CoreOS. Simple task to get you started.

Addressed by 29b71b8

### CODE REVIEW

**Comment by gitschaub** [comment]
Just realized that I omitted two items:

1) Check if previously bootstrapped. Because it's not a single script, should we do a check in main.yml before including coreos.yml?

2) Setting up http proxy for wget when downloading the pypy tarball. pxe-cloud-config.yml needs this information as well, is there a way we can centralize this configuration?

File: ansible/roles/pre-ansible/tasks/coreos.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@
   ignore_errors: True
 
 - name: CoreOS | Download pypy source
-  raw: wget -O /tmp/pypy-{{ pypy_version }}-linux64.tar.bz2 {{ pypy_base_url }}pypy-{{ pypy_version }}-linux64.tar.bz2
+  raw: wget -e http_proxy={{ http_proxy|default('') }} -e https_proxy={{ https_proxy|default('') }} -O /tmp/pypy-{{ pypy_version }}-linux64.tar.bz2 {{ pypy_base_url }}pypy-{{ pypy_version }}-linux64.tar.bz2
   when: need_pypy_tar
```
Comment (danehans) [inline ansible/roles/pre-ansible/tasks/coreos.yml]: Can you cherry-pick this commit into a topic branch sourced from [1]. Then submit a separate PR so pypy downlaod includes no_proxy support. This is b/c I eventually want to host the pypy tar ball on the gem-master. Feel free to also add pypy download to [2].

[1] https://github.com/danehans/contrib/tree/no_proxy
[2] https://github.com/danehans/ansible-role-ansible-controller

**Comment by danehans** [comment]
@adamschaub let me test this out first and then we'll see if it needs to be refined.

**Comment by danehans** [comment]
@adamschaub For 1, I'm not worried about the check. It sounds like eparis prefers we don;t implement checks as it changes Ansible from declarative to imperative. We can simply run the bootstrap on every ansible run. For 2, we need to implement the proxy/no_proxy stuff separately until we merge the PXE components upstream. I have stepped away from automating the gem-master for the last week to focus on other things, but I believe I use the same proxy/no_proxy logic from the existing code to set this. Therefore, setting proxy/no_proxy in group_vars hits the gem-master cloud-init file. If I did not include this, it will be easy to add.

**Pull request closed without merging**